### DISCUSSION TITLE
Send crash analytics via email or other alternatives

### DISCUSSION THREAD

**Discussion by nitanmarcel** [AvaloniaUI/Avalonia#6114] [Q&A]
So, I remember seeing for wpf a project that allowed developer to integrate a feature that allowed them to receive emails will crash statistics every time the app crashes.

Is there an way to do this using Avalonia? Either by using a separate library or a build-in one?

**Comment by maxkatz6** [comment]
Yes, use separate libraries.
Avalonia is GUI framework, it does not provide anything special about crash reporting.
It could be some kind of Logger Sink (i.e. serilog with sending logs to the server) or direct HTTP requests to the telemetry service (Google Analytics, App Center, Application Insigths), or wrapper on those telemetry services (Google Analytics has one, App Center has but not compatible with cross platform applications).
So it really depends on the library, but not on the Avalonia. It's the same as adding telemetry to the console app.
Only one major difference between Avalonia and WPF in that context, is the fact of non existing "global exception handled" in Avalonia. So no need to support anything Avalonia specific.

**Reply by Sorien** [reply]
we are using https://sentry.io, not for avalonia based app but works very well...
